azure-keyvault/examples/test_examples_keyvault.py

```diff
@@ -0,0 +1,197 @@
+#-------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+#--------------------------------------------------------------------------
+
+import pytest
+import os
+
+from azure.keyvault.secrets._client import SecretClient
+from azure.common.credentials import ServicePrincipalCredentials
+from azure.core.exceptions import ClientRequestError
+
+def create_vault_client():
+    # [START create_vault_client]
+    import os
+    from azure.keyvault.vault_client import VaultClient
+
+    vault_url = 'https://myvault.vault.azure.net.'
+    credentials =  ServicePrincipalCredentials(
+        client_id = client_id,
+         secret = client_secret,
+        tenant = tenant_id,
+        resource = "https://vault.azure.net"
+    )
+
+    # Create a new Vault client using Azure credentials
+    vault_client = VaultClient(vault_url=vault_url, credentials=credentials)
+    # [END create_vault_client]
+    return vault_client
+
+
+def test_example_secret_crud_operations():
+   
+    vault_client = create_vault_client()
+    client = vault_client.secrets
```

Comment:
Something to consider: is it preferable to show secret operations via `vault_client`, i.e. `vault_client.secrets.get_secret(...)`?

Reply:
hmm not sure, would a user prefer to write it as `vault_client.secrets.get_secret()` pr just `client.get_secret()`. Assuming the user has already created the secrets_client.

Reply:
I wouldn't demo using the VaultClient here, but rather I would use the SecretClient directly.  For customers just looking to use secret operations I think this is the most straight forward way to create like secret = SecretClient(url, creds)

---
In reply to: [276353128](https://github.com/chlowell/azure-sdk-for-python/pull/1#discussion_r276353128) [](ancestors = 276353128)

Reply:
Updated to use the Secret Client directly.
[here](https://github.com/samvaity/azure-sdk-for-python/blob/sv_secrets_docs_test/azure-keyvault/tests/test_examples_keyvault.py#L40)